compiler/capnpc-c.c

```diff
@@ -449,6 +449,8 @@ static const char *xor_member(struct field *f) {
 			return strf(&buf, " ^ %#xu", (uint32_t) f->v.intval);
 
 		case Value_int64:
+			return strf(&buf, " ^ ((int64_t)((uint64_t) %#x << 32) ^ %#x)",
+						(uint32_t) (f->v.intval >> 32), (uint32_t) f->v.intval);
 		case Value_uint64:
```

Comment:
There should be a "u" after `%#x` to make sure the constant is unsigned, otherwise the shift might be undefined.  Not sure why the code splits the 64bit constant into 2 32-bit pieces...